### PR TITLE
Fix system.part choice for Gibbs sample

### PR DESCRIPTION
**Pull request by jhossbach**
Fixes / Avoids #4278 causing a RuntimeError if a box runs empty

### CODE REVIEW

File: samples/gibbs_ensemble/gibbs_ensemble.py [hunk 1]
```diff
@@ -80,7 +80,9 @@ def send_info(self):
 
     def move_particle(self, DX_MAX):
         """ Move random particle inside the box """
-        self.old_particle = np.random.choice(self.system.part)
+        self.random_particle_id = np.random.choice(self.system.part[:].id)
+        self.old_particle = self.system.part[self.random_particle_id]
+        self.random_particle_id = None
```
Comment (jngrad) [inline samples/gibbs_ensemble/gibbs_ensemble.py]:
```suggestion
        random_particle_id = np.random.choice(self.system.part[:].id)
        self.old_particle = self.system.part[random_particle_id]
```

File: samples/gibbs_ensemble/gibbs_ensemble.py [hunk 2]
```diff
@@ -104,7 +106,9 @@ def revert_add_particle(self):
 
     def remove_particle(self):
         """ Remove a random particle """
-        self.old_particle = np.random.choice(self.system.part).to_dict()
+        self.random_particle_id = np.random.choice(self.system.part[:].id)
+        self.old_particle = self.system.part[self.random_particle_id].to_dict()
+        self.random_particle_id = None
```
Comment (jngrad) [inline samples/gibbs_ensemble/gibbs_ensemble.py]:
```suggestion
        random_particle_id = np.random.choice(self.system.part[:].id)
        self.old_particle = self.system.part[random_particle_id].to_dict()
```

**Review by jngrad — CHANGES_REQUESTED** [review]
Besides requested changes, looks good to me.

File: samples/gibbs_ensemble/gibbs_ensemble.py [hunk 3]
```diff
@@ -80,7 +80,8 @@ def send_info(self):
 
     def move_particle(self, DX_MAX):
         """ Move random particle inside the box """
-        self.old_particle = np.random.choice(self.system.part)
+        self.random_particle_id = np.random.choice(self.system.part[:].id)
+        self.old_particle = self.system.part[self.random_particle_id]
```
Comment (jngrad) [inline samples/gibbs_ensemble/gibbs_ensemble.py]: why do we need to store the `random_particle_id` as a member of the object if it is not used elsewhere?

Comment (jhossbach) [reply samples/gibbs_ensemble/gibbs_ensemble.py]: Looks like I wrote that out of habit, I'll force update it not to store it as a member.

**Review by jngrad — APPROVED** [review]

**Pull request merged**